src/routes/abstract-indexation-route.php

```diff
@@ -36,15 +57,24 @@ protected function respond_with( $objects, $next_url ) {
 	 * @param string                      $url               The url of the indexation route.
 	 *
 	 * @return WP_REST_Response The response.
+	 *
+	 * @throws Exception If the indexation action fails.
 	 */
 	protected function run_indexation_action( Indexation_Action_Interface $indexation_action, $url ) {
-		$indexables = $indexation_action->index();
+		try {
+			$indexables = $indexation_action->index();
+
+			$next_url = false;
+			if ( \count( $indexables ) >= $indexation_action->get_limit() ) {
+				$next_url = \rest_url( $url );
+			}
 
-		$next_url = false;
-		if ( \count( $indexables ) >= $indexation_action->get_limit() ) {
-			$next_url = \rest_url( $url );
+			return $this->respond_with( $indexables, $next_url );
 		}
+		catch ( Exception $exception ) {
+			$this->options_helper->set( 'indexables_indexation_reason', Indexing_Notification_Integration::REASON_INDEXING_FAILED );
```

Comment:
Having this here is way too specific. This class represents an abstract indexation route class and this line is about an implementation of this class.